### ISSUE TITLE
Method invocation via an interface can fail for ToString

### ISSUE THREAD

**Issue by c0pperdragon** [bridgedotnet/Bridge#3907]
This issue is pretty complicated. It arises when an interface unnecessarily declares the ToString method and it is then used in an even more complicated pattern of inheritance. I have broken down the example to the smallest possible program that still causes the issue.

### Steps To Reproduce

https://deck.net/612386e4dfd17ebdf7129056211083df

https://dotnetfiddle.net/GHQ0r3

```csharp
public class Test
{
    public static void Main()
    {
        I i = new B();
        
        // fails - calls wrong method  
        System.Console.WriteLine(i.ToString());  
    }
    
    public interface I
    {
        // this unnecessary declaration causes the problem
        string ToString();
    }
    
    public class A : I
    {
        public override string ToString()
        {
            return "A";
        }
    }
    
    public class B : A, I
    {
    }
}
```

### Expected Result

```js
A
```

### Actual Result

```js
[object Object]
```

**Comment by geoffreymcgill** [comment]
Interesting problem. I'm pretty sure we will be able to fix this scenario and include the fix in the upcoming Bridge 17.8.0 release.

**Comment by fabriciomurta** [comment]
The fix for this issue is implemented and works just fine! It will be included the coming Bridge release, version matching the assigned milestone.
